app/bundles/CampaignBundle/Controller/Api/EventLogApiController.php

```diff
@@ -27,20 +35,26 @@ class EventLogApiController extends CommonApiController
      */
     protected $contact;
 
-    /** @var EventLogModel */
-    protected $model;
+    /**
+     * @var EventLogModel|null
+     */
+    protected $model = null;
 
     public function initialize(ControllerEvent $event)
     {
-        $this->model                    = $this->getModel('campaign.event_log');
-        $this->entityClass              = 'Mautic\CampaignBundle\Entity\LeadEventLog';
+        $campaignEventLogModel = $this->getModel('campaign.event_log');
+        if (!$campaignEventLogModel instanceof EventLogModel) {
+            throw new RuntimeException('Wrong model given.');
+        }
+        $this->model                    = $campaignEventLogModel;
+        $this->entityClass              = LeadEventLog::class;
         $this->entityNameOne            = 'event';
         $this->entityNameMulti          = 'events';
         $this->parentChildrenLevelDepth = 1;
         $this->serializerGroups         = [
             'campaignList',
             'ipAddressList',
-            'log' => 'campaignEventLogDetails',
+            self::LOG_SERIALIZATION => 'campaignEventLogDetails',
```

Comment:
Serializer expects groups to be `array<int, string>` (sure it will work with string keys, but the code there is expecting array to be indexed, not associative. This change ensures the `campaignEventLogDetails` group is last in groups list.